guides/echo-guides/otp.mdx

```diff
@@ -0,0 +1,319 @@
+    fontSize: "15px",
+    fontFamily: "HelveticaNeue,Helvetica,Arial,sans-serif",
+    letterSpacing: "0",
+    lineHeight: "23px",
+    padding: "0 40px",
+    margin: "0",
+    textAlign: "center" as const,
+};
+
+const link = {
+    color: "#444",
+    textDecoration: "underline",
+};
+
+const codeContainer = {
+    background: "rgba(0,0,0,.05)",
+    borderRadius: "4px",
+    margin: "16px auto 14px",
+    verticalAlign: "middle",
+    width: "280px",
+};
+
+export function renderOTPEmail(payload: any) {
+    return render(<LinearLoginCodeEmail {...payload} />);
+}
+```
+
+2. Launch Dev Studio - Novu’s code-first approach lets you see how the template would look when rendered, right when defining it, using the Dev Studio. To launch the dev studio locally you can run npx novu-labs@latest echo. The Dev Studio will be started by default on port 2022, and accessible via: http://localhost:2022
+
+<Frame caption="Dev Studio lets you see the workflow you wrote without sending it off!"><img src="/images/echo-otp/studio.png" /> </Frame>
+
+3. Defining a workflow that uses that template to send notifications - In this step, we need to define a workflow that uses the template we wrote above to render the email notification:
+```ts
+import { Echo } from "@novu/echo";
+import { renderOTPEmail } from "./otp";
+
+export const echo = new Echo({
+  apiKey: '<NOVU_API_KEY>',
+  /**
+   * Enable this flag only during local development
+   */
+
+  // backendUrl: 'https://api.novu.co',
+  devModeBypassAuthentication: process.env.NODE_ENV === "development",
+});
+
+const otpFlow = echo.workflow('otp-flow', async ({ step, payload }) => {
+  // Send a welcome email
+  await step.email('send-email', async (inputs) => {
+    return {
+      subject: `Here's your verification code, Sumit!`,
+      body: renderOTPEmail(inputs, payload),
+    };
+  }, {
+    inputSchema: {
+      type: "object",
+      properties: {
+        showJoinButton: { type: "boolean", default: true },
+        buttonText: { type: "string", default: "Login to Linear" },
+        logoURL: {
+          type: "string",
+          default: "https://react-email-demo-7qy8spwep-resend.vercel.app/static/linear-logo.png",
+          format: "uri",
+        },
+        supportEmail: { type: "string", default: "support@linear.co" },
+        validationCode: { type: "string", default: "tt226-5398x" },
+        inviteLink: {
+          type: "string",
+          default: "https://linear.app",
+          format: "uri",
+        },
+        inviteFromIp: { type: "string", default: "204.13.186.218" },
+        inviteFromLocation: {
+          type: "string",
+          default: "São Paulo, Brazil",
+        },
+      },
+    },
+  });
+  // JSON Schema for validation and type-safety. Zod, and others coming soon.
+}, { payloadSchema: { properties: { text: { type: 'string' } } } });
+```
+4. Establishing the route - Lastly, we need a route that triggers the workflow when it is hit. Here’s the route we’re using:
```

Comment:
What route is established here? This seems confusing to me. The sentence says "here is the route we are using" but I don't see any route. All i see is the block of code in a file that triggers a notification.

Can it just be "Trigger a notification" ?

Reply:
Good point! Rephrasing it